controller/lib/hal/hal.h

```diff
@@ -235,9 +251,19 @@ class HalApi {
 
 extern HalApi Hal;
 
-#ifdef AVR
+#if defined(ARDUINO_AVR_UNO) || defined(ARDUINO_NUCLEO_L452RE)
 
 inline void HalApi::init() {
+  // We need to initialize Watchdog ASAP for two reasons:
+  //
+  //  - The purpose of the watchdog is to catch hangs, and if initializing the
+  //    watchdog weren't the first thing we did, we wouldn't catch hangs in the
+  //    work that came before.
+  //
+  //  - After the device is soft-reset via reset_device(), the watchdog timer
+  //    has a very short value.  We need to Hal.init() immediately so that
+  //    we don't time out while initializing.
```

Comment:
Not wild about copy-pasting the comment (same reason as not wild about copy-pasting code).  I think I see why you did it; the comment *is* sort of relevant in both places.  We need to know in `main` to initialize the watchdog first, and we need to know here in init() to do the same, initialize the watchdog first.

To reduce the duplication, what if we

 - moved this comment into HalApi::init(), saying "Hal.init() should be one of the first things called during initialization.  Among other things, it initializes the watchdog, and that needs to happen very early in startup because: ..."
 - In main.cpp, say "Initialize Hal first because it initializes the watchdog.  See comment on HalApi::init()."

?